xarray/plot/plot.py

```diff
@@ -555,14 +563,14 @@ def _infer_interval_breaks(coord):
 
 
 @_plot2d
-def pcolormesh(x, y, z, ax, **kwargs):
+def pcolormesh(x, y, z, ax, infer_intervals=True, **kwargs):
     """
     Pseudocolor plot of 2d DataArray
 
     Wraps matplotlib.pyplot.pcolormesh
     """
 
-    if not hasattr(ax, 'projection'):
+    if infer_intervals:
```

Comment:
I have mixed feelings about this change. I'm not sure I like `infer_intervals` to default to `True`. This is going to break a bunch of our plotting code. My preference, and I'm open to other ideas at this point, would be to default to `infer_intervals=None` with something like this logic:

```Python
if (infer_intervals is None) and (hasattr(ax, 'projection'))
    infer_intervals = False
    warnings.warn("infer_intervals=None is deprecated, please specify either True or False",
                  DeprecationWarning)
if infer_intervals:
    ...
```